### PR TITLE
BK-6129: Trace timing info to Jaeger

### PR DESCRIPTION
**Pull request by spellr**

### CODE REVIEW

**Comment by spellr** [comment]
jenkins test this

File: hammock/api.py [hunk 1]
```diff
@@ -8,9 +11,10 @@
 class Hammock(object):
     def __init__(
         self, api, resource_package,
-        policy_file=None, credentials_class=None, backend_args=None,
+        policy_file=None, credentials_class=None, backend_args=None, service_name=None, enable_tracing=True,
         **resource_params
     ):
+        service_name = service_name or 'hammock-service'
```
[the remark this answers is not in the document]
Comment (sharbov) [reply hammock/api.py]: maybe we should use the same environment variable I used,

you can even use my method to set the service name, see
https://github.com/Stratoscale/hammock/blob/master/hammock/templates/client/file.j2#L231

File: hammock/templates/client/file.j2 [hunk 2]
```diff
@@ -107,10 +113,7 @@ class {{ class_name }}(object):
             json.update(kwargs)
         if json and not isinstance(json, list):
             json = {key: value for key, value in six.iteritems(json) if value != NOT_SET}
-        _kwargs = {
-            "timeout": self._timeout,
-            "stream": True,
-        }
+        _kwargs = {"headers": self._client.headers}
```
Comment (sharbov) [inline hammock/templates/client/file.j2]: why did you remove the `stream` keyword ?

File: hammock/tracing/current_span.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from __future__ import absolute_import
```
Comment (sharbov) [inline hammock/tracing/current_span.py]: can you explain what is this for?

File: hammock/tracing/client.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import six
+import opentracing
+from opentracing import Format
+from opentracing.ext import tags
+
+from hammock.tracing.current_span import get_current_span
+
+
+def before_request(request):
```
Comment (sharbov) [inline hammock/tracing/client.py]: maybe you can use this inside the context manager, instead of requiring the user to do so for you

**Comment by sharbov** [comment]
nice work,

I do however recommend a few changes:

* implement `before` & `after` hooks in both server & client that will be used by configurable plugins.
* pass the tracing plugin as a dependency to the constructor, avoiding import management.
* plugins should implement a set interface of before and after methods.

I hope you'll apply these changes.

good luck

File: hammock/tracing/current_span.py (hunk 3 above)
Comment (sharbov) [reply hammock/tracing/current_span.py]: consider adding this nice explanation to a doc string

File: hammock/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (spellr) [reply hammock/api.py]: @sharbov I missed that code in hammock client, but I really dislike reading strato-log from inside hammock, which is open-source.
I understand that no one outside stratoscale currently uses hammock, and probably never will. But if Stratoscale has decided it wants to contribute to open-source, we need to do that the right way.
I don't mind reading the environment variable. I've added that. Getting the process name is a bit more complicated and probably requires adding a requirement to hammock
